main.bal

```diff
@@ -1,56 +1,128 @@
+    byte[] buffer2 = [];
+    int? expectedLen = ();
+    int? lengthOne = ();
+    int? lengthTwo = ();
+    int x = 0;
+    string message = "";
 
-            if firststring is string{
-                string finalstring = regex:replaceAll(firststring, "/[^ -~]+/g", "");
-                int x = <int> finalstring.indexOf("{");
-                int y = finalstring.length();
-                string secondstring = finalstring.substring(x,y);
-                
-                // io:println(secondstring);
+    function handleInput(tcp:Caller caller, readonly & byte[] data) returns error? {
 
-                io:Error? fileWriteString2 = io:fileWriteString("file7.json",secondstring,io:APPEND);
+        if self.expectedLen == () {
 
-                // json|error j = secondstring.fromJsonString();
-                // io:println(j);
+            self.buffer = data;
+            self.expectedLen = check extractContentLength(data);
+            self.lengthOne = self.contentLengthRemover(self.buffer).length();
 
+        } 
+        else {
+
+            self.buffer2.push(...data);
+            string|error stringTwo = string:fromBytes(self.buffer2);
+            if stringTwo is string {
+                self.lengthTwo = stringTwo.length();
             }
+        }
+
+        int stringLength = <int>self.lengthOne + <int>self.lengthTwo;
+        if (self.expectedLen == stringLength) {
+
+            self.buffer.push(...self.buffer2);
+            json jsonObject = self.contentLengthRemover(self.buffer).toJson();
+            io:println(jsonObject);
+
+            if (self.x == 0) {
+                self.message = "Content-Length: 171\r\n\r\n{\"jsonrpc\":\"2.0\",\"id\":0,\"result\":{\"capabilities\":{\"textDocumentSync\":2,\"completionProvider\":{\"resolveProvider\":true},\"workspace\":{\"workspaceFolders\":{\"supported\":true}}}}}";
+                self.x += 1;
+            } 
+            else if (self.x == 1) {
+                self.message = "Content-Length: 201\r\n\r\n{\"jsonrpc\":\"2.0\",\"id\":0,\"method\":\"client/registerCapability\",\"params\":{\"registrations\":[{\"id\":\"9716545a-b5e9-4b32-94e1-be674a62eb04\",\"method\":\"workspace/didChangeConfiguration\",\"registerOptions\":{}}]}}";
+                self.x += 1;
+            } 
+            else if (self.x == 2) {
+                self.message = "Content-Length: 204\r\n\r\n{\"jsonrpc\":\"2.0\",\"id\":1,\"method\":\"client/registerCapability\",\"params\":{\"registrations\":[{\"id\":\"36b409a1-be16-4010-b7bf-a26b6f659596\",\"method\":\"workspace/didChangeWorkspaceFolders\",\"registerOptions\":{}}]}}";
+                self.x += 1;
+            } 
+            else if (self.x == 3) {
+                self.message = "Content-Length: 177\r\n\r\n{\"jsonrpc\":\"2.0\",\"id\":2,\"method\":\"workspace/configuration\",\"params\":{\"items\":[{\"scopeUri\":\"file:///c%3A/Users/Mindula/Desktop/vext/out.txt\",\"section\":\"languageServerExample\"}]}}";
+                self.x += 1;
+            } 
```

Comment:
This logic has to be generified to `n` steps, Please put the messages in an array, as a module level variable (outside the function), shouldn't need lot of ifs (zero if should be possible here)